### ISSUE TITLE
Publish menu available for all project types

### ISSUE THREAD

**Issue by mrward** [mono/monodevelop#6465]
VSMac: 7.7.0.1750
Git revision: 743007e584bcc371b4f8314c199cfce858ebc7da
Build date: 2018-10-24 17:03:24+00
Build branch: release-7.7
Xamarin extensions: 4eb7bc2cc6d74dadc5b397d8c5c13f334bc258b3

The Publish menu is available for all project types. If you try to publish in a non .NET Core project, such as a C# console project, it will fail.

```
dotnet publish
Microsoft (R) Build Engine version 15.7.179.6572 for .NET Core
Copyright (C) Microsoft Corporation. All rights reserved.

  Nothing to do. None of the projects specified contain packages to restore.
/usr/local/share/dotnet/sdk/2.1.302/Microsoft.Common.CurrentVersion.targets(1179,5): error MSB3644: The reference assemblies for framework ".NETFramework,Version=v4.7" were not found. To resolve this, install the SDK or Targeting Pack for this framework version or retarget your application to a version of the framework for which you have the SDK or Targeting Pack installed. Note that assemblies will be resolved from the Global Assembly Cache (GAC) and will be used in place of reference assemblies. Therefore your assembly may not be correctly targeted for the framework you intend. [test.csproj]
The application was terminated by a signal: SIGHUP

dotnet publish returned: 1
```

The Publish menu should only be visible/enabled for .NET Core projects.

Also the Publish menu is available for every item in the Solution pad. Right clicking a file or a folder shows the Publish menu. Not sure if that is expected since you cannot publish an individual file or folder presumably.

> VS bug [#715107](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/715107)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.8.0.1038 (master)

Pull Request #6466 merged by: Jose Miguel Torres
Author: mono
Commit: 0bc0954ad91c133efb28ef8d2b1269c0638dadbd (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 7.7.0.1776 (release-7.7)

Pull Request #6473 merged by: Lluis Sanchez
Author: mono
Commit: d387b70696545900297d9e115240fa7a4fadbec7 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-10] Comment by **xamarinc**



Fixed in version 7.7.0.1776 (release-7.7)



Pull Request #6473 merged by: Lluis Sanchez  
 Author: mono  
 Commit: d387b70696545900297d9e115240fa7a4fadbec7 (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.0.0.509 (release-8.0)

Pull Request #6466 merged by: Jose Miguel Torres
Author: mono
Commit: 0bc0954ad91c133efb28ef8d2b1269c0638dadbd (mono/monodevelop)

**Comment by xamarin-release-manager** [comment]
Fixed in version 8.6.0.802 (master)

Pull Request #6466 merged by: Jose Miguel Torres
Author: mono
Commit: 0bc0954ad91c133efb28ef8d2b1269c0638dadbd (xamarin/monodevelop)

**Comment by xamarin-release-manager** [comment]
> [sync] [VS-18] Comment by **xamarinc**



Fixed in version 8.6.0.802 (master)



Pull Request #6466 merged by: Jose Miguel Torres  
 Author: mono  
 Commit: 0bc0954ad91c133efb28ef8d2b1269c0638dadbd (xamarin/monodevelop)
